storage/pebble/blocks.go

```diff
@@ -253,6 +253,15 @@ func (b *Blocks) getBlock(keyCode byte, key []byte) (*models.Block, error) {
 	}
 
 	if b.chainID == flowGo.Testnet && slices.Contains(testnetBrokenParentHashBlockHeights, block.Height) {
+		// Since we are going to modify the `block.ParentBlockHash` field,
+		// we need to set the `block.FixedHash` field. If we don't do so,
+		// `block.Hash()` will return a different hash.
+		blockHash, err := block.Hash()
+		if err != nil {
+			return nil, err
+		}
+		block.FixedHash = blockHash
```

Comment:
Can we move all the special cases to a separated files? Otherwise, we would have to carry this field for future blocks who doesn't need it.

Reply:
It may be somewhat misleading, but `FixedHash` here does not refer to some pre-calculated hash value, that is hard-coded somewhere. It refers to the fact that the `Block` type, does not have a `Hash` field, it only has a `Hash` method, and it is dependent on the fields of the `Block` type, which can change (added/removed).

Reply:
Maybe I missed something, but why a block can be changed? We could use a different type if changes are needed , and when changes are done, create this immutable block, where Hash is correct.